install/kubernetes/helm/subcharts/gateways/templates/deployment.yaml

```diff
@@ -54,6 +54,19 @@ spec:
             privileged: true
 {{- end }}
       containers:
+{{- if eq $key "istio-ingressgateway" }}
+{{- if $spec.ingressSds.enabled }}
+        - name: ingress-sds
+{{- if contains "/" $spec.ingressSds.image }}
+          image: "{{ $spec.ingressSds.image }}"
+{{- else }}
+          image: "{{ $.Values.global.hub }}/{{ $spec.ingressSds.image }}:{{ .Values.global.tag }}"
+{{- end }}
+          volumeMounts:
+          - mountPath: /var/run/ingress_gateway/uds_path
```

Comment:
Should we use the same path for all 'sds' ? And please don't name the file 'uds_path' - you can use 'sds'.

Reply:
Correct me if I'm wrong. If we share the same SDS path for the ingress gateway agent and the Citadel agent, these agents need to differentiate the messages, which might be tricky. More importantly, that will eliminate the benefit of decoupling the gateway agent from the Citadel agent, right? A malicious node agent can just answer the request for gateway key/cert.